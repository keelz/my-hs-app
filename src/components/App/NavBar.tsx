import * as React from 'react';
import { Link } from 'react-router-dom';

interface AppNavBarProps {}

const AppNavBar: React.SFC<AppNavBarProps> = props =>
  <React.Fragment>
    <Link to="/" style={{ paddingRight: 10 }}>Home</Link>
    <Link to="/test">Test</Link>
  </React.Fragment>;

export default AppNavBar;
