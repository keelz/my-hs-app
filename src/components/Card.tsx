import * as React from 'react';
import * as classnames from 'classnames';
import { composeAssetSource } from '../common/services/hsJsonApi';
import {
  CardExt,
  CardLocale,
  CardResolution
} from '../common/models/Card';

interface ICardProps {
  className?: string | string[];
  ext: CardExt;
  id: string;
  locale: CardLocale;
  resolution: CardResolution;
}

const composeClassnames = (className?: string | string[]) =>
  classnames(
    [
      'Card',
    ],
    className
  );

const Card: React.SFC<ICardProps> = props =>
  <img
    className={composeClassnames(props.className)}
    src={composeAssetSource(
      props.id,
      props.locale,
      props.resolution,
      props.ext
    )} />;

export default Card;
