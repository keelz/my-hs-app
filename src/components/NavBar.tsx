import * as React from 'react';
import * as classnames from 'classnames';
import { Link } from 'react-router-dom';

interface INavBarProps {
  id: string;
  targets: ITarget[];
}

export interface ITarget {
  action?: (...args: any[]) => void;
  path: string;
  text?: string;
}

const composeClassnames = () => classnames([
  'Nav-bar',
]);

const NavBar: React.SFC<INavBarProps> = props =>
  <div id={props.id} className={composeClassnames()}>
    { props.targets.map((target, i) =>
        <Link
          key={`nav-bar-link-${props.id}-${i}`}
          onClick={target.action}
          className="link"
          to={target.path}>
          { target.text
              ? <span>{target.text}</span>
              : <></>
          }
        </Link>
      )
    }
  </div>;

export default NavBar;
