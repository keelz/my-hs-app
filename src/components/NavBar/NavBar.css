.Nav-bar {
  display: flex;
  justify-content: center;
}

.Nav-link {
  margin-right: .3125rem;
}
