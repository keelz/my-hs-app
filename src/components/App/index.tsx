import * as React from 'react';
import App from './App';
import LoadingAnimation from '../LoadingAnimation';

// redux props.
type IAppStateProps = {
  loaded: boolean;
  appDidLoadAction: () => any;
};

class Index extends React.Component<IAppStateProps> {
  componentDidMount() {
    this.props.appDidLoadAction();
  }

  render() {
    return (
      <div className="App container">
        { this.props.loaded
          ? <App />
          : <LoadingAnimation />
        }
      </div>
    );
  }
}

export default Index;
