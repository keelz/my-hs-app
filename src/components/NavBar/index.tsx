import * as React from 'react';
import { default as NavLinkComponent, NavLinkProps } from './NavLink';
import './NavBar.css';

// implementation props
interface INavBarProps {
  id: string;
}

export const NavBar: React.SFC<INavBarProps> = props =>
  <div id={props.id} className="Nav-bar">
    { props.children }
  </div>;

export const NavLink: React.SFC<NavLinkProps> = props =>
  <NavLinkComponent {...props}>{props.children}</NavLinkComponent>;
