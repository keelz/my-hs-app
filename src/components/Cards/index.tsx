import * as React from 'react';
import Card from '../Card';
import './Cards.css';
import {
  Card as CardType,
  CardResolution,
  CardExt,
  CardLocale
} from '../../redux/reducers/Cards';

interface CardsProps {}

export interface CardsStateProps {
  activeCardClassName: string;
  cards: CardType[];
}

type Props = CardsProps & CardsStateProps;

const Cards: React.SFC<Props> = props =>
  <div className="Cards">
    <div className="Cards-title">
      {props.activeCardClassName}: {props.cards.length}
    </div>
    <div className="Cards-collection">
      { props.cards.map(card =>
        <Card
          key={card.id}
          ext={CardExt.PNG}
          id={card.id}
          locale={CardLocale.EN}
          resolution={CardResolution.SMALL} />)
      }
    </div>
  </div>;

export default Cards;
