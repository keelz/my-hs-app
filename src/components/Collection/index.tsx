import * as React from 'react';
import Card from '../Card';
import './Collection.css';
import { ICollection } from '../../common/models/Collection';
import ManaBar from '../../redux/containers/Collection/ManaBar';
import {
  CardResolution,
  CardExt,
  CardLocale,
  CardClassName
} from '../../common/models/Card';

export interface ICollectionStateProps {
  activeCardClassName: CardClassName;
  collection: ICollection;
}

const Collection: React.SFC<ICollectionStateProps> = props =>
  <div className="Collection">
    <ManaBar />
    <div className="Collection-body">
      { props.collection.cards.map(card =>
        <Card
          key={card.id}
          ext={CardExt.PNG}
          id={card.id}
          locale={CardLocale.EN}
          resolution={CardResolution.SMALL} />)
      }
    </div>
  </div>;

export default Collection;
