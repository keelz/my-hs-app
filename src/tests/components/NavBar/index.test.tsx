import * as React from 'react';
import * as enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NavBar } from '../../../components/NavBar';

const composeProps = () => ({
  id: 'test',
  targets: [{
    path: '/test',
    text: 'test',
  }],
});

describe('NavBar', () => {
  describe('snapshots', () => {
    it('renders without crashing', () => {
      const tree = enzyme.shallow(<NavBar {...composeProps()} />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });

    it('renders correctly without text', () => {
      const props = { ...composeProps() };
      props.targets = props.targets.map(t => {
        delete t.text;
        return t;
      });
      const tree = enzyme.shallow(<NavBar {...props} />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });
  });
});
