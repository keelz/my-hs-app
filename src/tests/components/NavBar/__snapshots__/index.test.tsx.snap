// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar snapshots renders correctly without text 1`] = `
<div
  className="Nav-bar"
  id="test"
/>
`;

exports[`NavBar snapshots renders without crashing 1`] = `
<div
  className="Nav-bar"
  id="test"
/>
`;
